src/index/StringSortComparator.h

```diff
@@ -0,0 +1,462 @@
+    explicit WeightString(std::string_view contents)
+        : _content(contents){}[[nodiscard]] const std::string & get() const {
+      return _content;
+    }
+    std::string& get() { return _content; }
+
+   private:
+    std::string _content;
+  };
+
+  // ____________________________________________________________________________________________________________
+  LocaleManager(const LocaleManager& rhs)
+      : _icuLocale(rhs._icuLocale),
+        _ignorePunctuationStatus(rhs._ignorePunctuationStatus) {
+    setupCollators();
+    setIgnorePunctuationOnFirstLevels(_ignorePunctuationStatus);
+  }
+
+  // Current usage requires default-constructibility. Default to "en_US" and
+  // respecting symbols on all the levels
+  LocaleManager() : LocaleManager("en", "US", false){};
+
+  /**
+   * @param lang The language of the locale, e.g. "en" or "de"
+   * @param country The country of the locale, e.g. "US" or "CA"
+   * @param ignorePunctuationAtFirstLevel If true then spaces/punctuation etc.
+   * will only be considered for comparisons if strings match otherwise Throws
+   * std::runtime_error if the locale cannot be constructed from lang and
+   * country args
+   *
+   * \todo: make the exact punctuation level configurable.
+   */
+  LocaleManager(const std::string& lang, const std::string& country,
+                bool ignorePunctuationAtFirstLevel) {
+    _icuLocale = icu::Locale(lang.c_str(), country.c_str());
+    _ignorePunctuationStatus =
+        ignorePunctuationAtFirstLevel ? UCOL_SHIFTED : UCOL_NON_IGNORABLE;
+
+    if (_icuLocale.isBogus() == TRUE) {
+      throw std::runtime_error("Could not create locale with language " + lang +
+                               " and Country " + country);
+    }
+    setupCollators();
+    setIgnorePunctuationOnFirstLevels(_ignorePunctuationStatus);
+  }
+
+  // ____________________________________________________________________
+  LocaleManager& operator=(const LocaleManager& other) {
+    if (this == &other) return *this;
+    _icuLocale = other._icuLocale;
+    _ignorePunctuationStatus = other._ignorePunctuationStatus;
+    setupCollators();
+    setIgnorePunctuationOnFirstLevels(_ignorePunctuationStatus);
+    return *this;
+  }
+
+  // _____________________________________________________________________
+  [[nodiscard]] int compare(std::string_view a, std::string_view b,
+                            const Level level) const {
+    UErrorCode err = U_ZERO_ERROR;
+    auto idx = static_cast<uint8_t>(level);
+    auto res = compToInd(
+        _collator[idx]->compareUTF8(toStringPiece(a), toStringPiece(b), err));
+    raise(err);
+    return res;
+  }
+
+  // _____________________________________________________________________
+  static int compare(WeightString a, WeightString b,
+                     [[maybe_unused]] const Level) {
+    return std::strcmp(a.get().c_str(), b.get().c_str());
+  }
+
+  // transform a UTF-8 string into a weight string that can be compared using
```

Comment:
Hm, I am currently working on the unification of the comments and am really wondering about the correct way.
Is it so bad to use `///` and `/**  */`? The first one of course then should only be used for a single line and is doxygen shorthand for a integrated `@brief`.
Also it sometimes is typical to use doxygen style `///` and `/** */` for public interfaces and simple C++ style
(`//` and `/* */`) which will be ignored by doxygen in the private interfaces.

Do you really think we should use `/** */` even for simple one line comments like "This is the Copy Constructor that copies as expected"? 
I am looking forward to hear your opinion about this.

Reply:
Hmm, I agree with you, as long as `///` is only used for one line comments a mix is fine. I'm currently in the process of getting used to only `/**/` for Linux kernel work so I might have been a bit sensitive because of that.

Reply:
Ok, then we are on the same level. Isn't the linux kernel written in `C` and the `//` comments are C++ comments which are supported by most C-Compilers but forbidden by the Ansi-C standard?

Reply:
@joka921 yes it is and yes `//` is C++ only. However even in C mode gcc doesn't complain about them so if one uses them by accident only the checkpatch script will actually complain.